addons/web/static/src/core/utils/strings.js

```diff
@@ -151,22 +151,20 @@ export function odoomark(text) {
 
 /**
  * Returns a markuped version of the input text where
- * the query is highlighted using the input classes and
- * a b tag if it is part of the text
+ * the query is highlighted using the input classes
+ * if it is part of the text.
  *
  * @param {string} query
- * @param {string} text
+ * @param {string | Markup} text
  * @param {string} classes
- * @returns {string}
+ * @returns {ReturnType<markup>}
  */
 export function highlightText(query, text, classes) {
     if (!query) {
-        return odoomark(text);
+        return htmlEscape(text);
```

Comment:
`htmlEscape` should never be necessary when used alone. It is only useful when about to "inject" into something (basically when injecting into HTML node directly, or when injecting into a markup() call).

This should either be removed, or it highlights a security bad practice in the rest of the code: code using `highlightText` assuming it is safe? In that case `htmlEscape` should be moved to call site of `highlightText` instead.